openapi-generator-annotations/test/openapi_generator_annotations_test.dart

```diff
@@ -28,30 +37,167 @@ void main() {
+            expect(InputSpec(path: 'path').path, 'path');
+          });
+        });
+        group('Remote Spec', () {
+          group('AWS', () {
+            group('defaults', () {
+              final remoteSpec = AwsRemoteSpec(
+                region: 'region',
+                bucket: 'bucket',
+                path: 'path',
+              );
+              test('has set values', () {
+                expect(remoteSpec.bucket, 'bucket');
+                expect(remoteSpec.region, 'region');
+                expect(remoteSpec.path, 'path');
+                expect(remoteSpec.accessKeyId, isNull);
+                expect(remoteSpec.secretAccessKey, isNull);
+              });
+              test('returns empty auth headers when access keys are null', () {
+                expect(remoteSpec.authHeaderContent, isEmpty);
+              });
+              test(
+                  'toHeaderMap returns empty when unable to use or find access keys',
+                  () {
+                expect(remoteSpec.toHeaderMap(), isEmpty);
+              });
+              // TODO: Find a way to cleanly update environment variables.
+              //  Isolate.spawnUri looks like a possible solution but I'm not sure
+              //  how to use it.
+              test('loadsCredentials', () {
+                final remoteSpec = AwsRemoteSpec(
+                  region: 'region',
+                  bucket: 'bucket',
+                  path: 'path',
+                );
+                expect(remoteSpec.accessKeyId, isNull);
+                expect(remoteSpec.secretAccessKey, isNull);
+                final map = <String, String>{};
+                map['AWS_ACCESS_KEY_ID'] = 'test';
+                map['AWS_SECRET_ACCESS_KEY'] = 'test';
+                Platform.environment.addAll(map);
+
+                Platform.environment.remove(map);
+                expect(remoteSpec.accessKeyId, 'test');
+                expect(remoteSpec.secretAccessKey, 'test');
+              }, skip: true);
+              test('toHeaderMapLoadsCredentials', () {
+                final remoteSpec = AwsRemoteSpec(
+                  region: 'region',
+                  bucket: 'bucket',
+                  path: 'path',
+                );
+                expect(remoteSpec.accessKeyId, isNull);
+                expect(remoteSpec.secretAccessKey, isNull);
+                Platform.environment['AWS_ACCESS_KEY_ID'] = 'test';
+                Platform.environment['AWS_SECRET_ACCESS_KEY'] = 'test';
+                final originalNow = remoteSpec.now;
+                final map = remoteSpec.toHeaderMap();
+                final newNow = remoteSpec.now;
+                Platform.environment.remove('AWS_ACCESS_KEY_ID');
+                Platform.environment.remove('AWS_SECRET_ACCESS_KEY');
+                expect(map['Authorization'], isNotNull);
+                expect(map['x-amz-date'], isNotNull);
+                expect(originalNow != newNow, isTrue);
+              }, skip: true);
```

Comment:
I'm not used to working with the isolates and the docs kinda suck around usage will play around with it.